File: Data Structures & Algorithms/subsets/submission-4.go
import "slices"
func subsets(nums []int) [][]int {
    ans := make([][]int,0)

    var rec func(int, []int)

    rec = func(i int, temp []int){
        if i == len(nums){
            t := make([]int, len(temp))
            copy(t, temp)
            ans = append(ans, t)
            return
        }

        // include
        temp = append(temp, nums[i])
        rec(i+1,temp)
        temp = slices.Delete(temp,len(temp)-1,len(temp))
        rec(i+1, temp)
    }
    rec(0,[]int{})

    return ans

}

File: Data Structures & Algorithms/combination-target-sum/submission-4.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:

        ans = []
        def rec(i, arr, cur):
            if i >= len(nums):
                return
            if cur > target:
                return
            if cur == target:
                ans.append(arr.copy())
                return

            arr.append(nums[i])
            rec(i,arr,cur+nums[i])
            arr.pop()

            rec(i+1,arr,cur)

        rec(0,[],0)
        return ans

            
        
        